project_cost_report/report/report.py

```diff
@@ -0,0 +1,320 @@
+        return babel.dates.format_date(now, "long", locale=lang)
+
+    def _get_product_categories(self, project, report_context):
+        """Get the stockable/consumable product categories."""
+        lines = self._get_product_analytic_lines(project)
+        default_cost_category = self.env.ref(
+            "project_cost_report.cost_category_product"
+        )
+        grouped_lines = group_analytic_lines(
+            lines,
+            lambda l: l.product_id.categ_id.project_cost_category_id
+            or default_cost_category,
+        )
+        sorted_categories = sorted(grouped_lines.keys(), key=lambda c: c.name)
+        unfolded_categories = report_context.get("unfolded_categories") or {}
+        unfolded_product_categories = unfolded_categories.get("product") or []
+        return [
+            CostReportCategory(
+                c,
+                lines=grouped_lines.get(c),
+                folded=c.id not in unfolded_product_categories,
+            )
+            for c in sorted_categories
+        ]
+
+    def _get_product_total(self, project):
+        lines = self._get_product_analytic_lines(project)
+        total_amount = sum(l.amount for l in lines)
+        return float_round(-total_amount, 2)
+
+    def _get_product_analytic_lines(self, project):
+        return project.analytic_account_id.line_ids.filtered(
+            lambda l: is_product_line(l)
+        )
+
+    def _get_time_categories(self, project, report_context):
+        """Get the task types for the TIME section."""
+        lines = self._get_timesheet_analytic_lines(project)
+        default_cost_category = self.env.ref("project_cost_report.cost_category_labour")
+        grouped_lines = group_analytic_lines(
+            lines,
+            lambda l: l.task_id.task_type_id.project_cost_category_id
+            or default_cost_category,
+        )
+        sorted_categories = sorted(grouped_lines.keys(), key=lambda c: c.name or "")
+        unfolded_categories = report_context.get("unfolded_categories") or {}
+        unfolded_time_categories = unfolded_categories.get("time") or []
+        return [
+            TimeCategory(
+                c,
+                lines=grouped_lines.get(c),
+                folded=c.id not in unfolded_time_categories,
+            )
+            for c in sorted_categories
+        ]
+
+    def _get_time_total(self, project):
+        lines = self._get_timesheet_analytic_lines(project)
+        total_amount = sum(l.amount for l in lines)
+        return float_round(-total_amount, 2)
+
+    def _get_time_total_hours(self, project):
+        lines = self._get_timesheet_analytic_lines(project)
+        total_hours = sum((l.unit_amount or 0) for l in lines)
+        return float_round(total_hours, 2)
+
+    def _get_timesheet_analytic_lines(self, project):
+        return project.analytic_account_id.line_ids.filtered(
+            lambda l: is_timesheet_line(l)
```

Comment:
Codacy found an issue: [Lambda may not be necessary](https://app.codacy.com/gh/Numigi/odoo-project-addons/pullRequest?prid=6165869)